### PR TITLE
Add signed uploads support

### PR DESCRIPTION
**Pull request by bautrukevich**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,6 +4,11 @@ All notable changes to this project will be documented in this file.
 The format is based now on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/)
 and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.html).
 
+## Unreleased
+
+* CDN Base option added.
+* Signed uploads added.
```
Comment (dmitry-mukhin) [inline CHANGELOG.md]:
```
### Added

* CDN Base option
* Signed uploads option
```

File: inc/utils.php [hunk 2]
```diff
@@ -183,14 +187,26 @@ function _uploadcare_get_js_cfg() {
...
+
+    $api = uploadcare_api();
+    $secureSignature = $api->widget->getSecureSignature();
+
+    if (!is_null($secureSignature)) {
+        return array_merge($baseParams, array(
+            'secureSignature' => $secureSignature->getSignature(),
+            'secureExpire' => $secureSignature->getExpire(),
+        ));
+    }
```
Comment (dmitry-mukhin) [inline inc/utils.php]: should we put it into `get_option('uploadcare_lifetime') > 0` condition?

File: inc/uploadcare_settings.php [hunk 3]
```diff
@@ -76,6 +78,7 @@
     $uploadcare_finetuning = get_option('uploadcare_finetuning');
     $uploadcare_source_tabs = get_option('uploadcare_source_tabs', $tab_defaults);
     $uploadcare_tab_effects = get_option('uploadcare_tab_effects', $effects_defaults);
+    $uploadcare_lifetime = get_option('uploadcare_lifetime', '0');
```
Comment (dmitry-mukhin) [inline inc/uploadcare_settings.php]:
```suggestion
    $uploadcare_lifetime = get_option('uploadcare_upload_lifetime', '0');
```

we'll probably want to add support for Authenticater URLs as well, with something like `uploadcare_cdn_lifetime` option name.

Or `uploadcare_upload_sig_lifetime` and `uploadcare_cdn_sig_lifetime` to make it more verbose.

File: inc/utils.php (hunk 2 above)
Comment (bautrukevich) [reply inc/utils.php]: yes, it would be good even add this check to uploadcare php, that lifetime is > 0.

File: inc/utils.php [hunk 4]
```diff
@@ -183,14 +187,26 @@ function _uploadcare_get_js_cfg() {
...
         'previewStep' => $previewStep,
         'effects' => implode(',', $effects),
         'ajaxurl' => admin_url('admin-ajax.php'),
         'tabs' => $tabs,
-        'cdnBase' => 'https://' . get_option('uploadcare_cdn_base', 'ucarecdn.com')
+        'cdnBase' => 'https://' . get_option('uploadcare_cdn_base', 'ucarecdn.com'),
     );
+
+    $api = uploadcare_api();
+    $secureSignature = $api->widget->getSecureSignature();
```
Comment (dmitry-mukhin) [inline inc/utils.php]: we don't need this, if signature is disabled (vast majority of cases)
that's why I ask to check the option here.

Comment (bautrukevich) [reply inc/utils.php]: Ok, I agree and understand now

**Pull request merged**